app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,32 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @post = @user.posts
+  end
+
+  def new
+    @post = Post.new
+
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text).merge(users: current_user, comments_counter: 0,
```

Comment:
- [ ] Please use `stong parameters` as a separate private method to follow SOLID development practices:
Create:
```
@post = Posts.new(post_params)
@post.users_id= @current_user
@post.comments_counter= 0
@post.likes_counter= 0
```

Stong params:
```ruby
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

- [ ] Please add these changes for other models as well 😉 


- [ ] OPTIONAL: In rails schema you have `users_id`, `posts_id` instead of `user_id`  `post_id`, having these reference in plural is wrong because: 
- You cannot create `user.post.create()`, this works only if user reference is in singular form. 
- Every time you have to explicitly point on id like `post.users_id = User.first.id`

I recommend you dropping the database, change migrations and run them again. Change migrations 
from: 
```
 def change
    add_reference :likes, :users, null: false, foreign_key: true
  end
```
To: 
```
 def change
    add_reference :likes, :user, null: false, foreign_key: true
  end
```
 
All the references should be in singular form.